### DISCUSSION TITLE
How to change wifi settings ?

### DISCUSSION THREAD

**Discussion by zigomatichub** [jomjol/AI-on-the-edge-device#2147] [Q&A] [answered]
I would like to update the wifi SSID and password to connect to another network. 
I don't find in config.ini. 
Is it possible to change that after the installation / when everything was setup  ?

**Comment by friedpa** [comment, marked as answer]
The WLAN Settings are stored in the WLAN.ini in the root folder. You must change the SSID/ Password directly on the SD-Card.

**Reply by Floppy-8D** [reply]
I know its old, but:
You can just upload via filemanager the new wlan.ini.
But it will fail, as it exists. So you have to delete it via post request like   http://<yourIP>/delete/wlan.ini
works with 15.7

**Reply by whitedavidp** [reply]
I am on v16 of jomjol. I would like to get, update, and deploy a new wifi config. But I am confused how to do this. Can you please elaborate? Thanks

**Reply by arneman** [reply]
This worked with v16.1:

1. Create a file wlan.ini: Open a text editor (like Notepad++ or VS Code) and create a new file named wlan.ini.
2. Add credentials: Enter your network data using the following format (including quotes, hostname is optional):
```
SSID = "Your_Network_Name"
password = "Your_New_Password"
hostname = "Your Hostname"
```

3. Set Encoding: Ensure the file is saved in UTF-8 (without BOM) to avoid connection errors.
4. Compress to ZIP: Create a ZIP archive (e.g., wifi_update.zip) containing only this wlan.ini file. Do not put it inside a folder within the ZIP.
5. Access Web-UI: Open your AI-on-the-Edge interface in your browser.
Upload via OTA:
Navigate to System in the menu.
Select OTA Update.
Click "Choose File" and select your ZIP file.
Click Update.
6. Reboot: The device will extract the file, overwrite the hidden configuration, and restart automatically.

**Comment by thomaslunze** [comment]
I would love to have a way to configure this without removing the sdcard again as most likely I will need to re-calibrate everything after touching the hardware. Would be really cool to get a way to edit/replace files or just the wlan.ini remotely.

**Comment by teixeluis** [comment]
I also share the same opinion. Maybe add basic authentication to the UI, but then also allow the wlan.ini file to be modified (or other form of updating the wifi settings). Huge hassle having to manually change it on the sd card. For example I am normally 170 Km away from my watermeter. I would like to change Wifi AP because the current one is a bit weak. I will have to wait for the next time I will be on site, open the device, take the sd card, put it back in place, realign everything. Just for changing the wifi settings.

**Comment by SybexX** [comment]
https://github.com/SybexX/AI-on-the-edge-device/actions/runs/10669878308

![Unbenannt](https://github.com/user-attachments/assets/beabfed7-cef7-488a-98b5-c0750a778613)

![Unbenannt1](https://github.com/user-attachments/assets/f6b553e0-6053-496b-b240-087f8e47d2da)

**Reply by SybexX** [reply]
Anyone interested is welcome to try it out and give feedback.

**Reply by turbolooser** [reply]
I have installed the update.zip but cannot find the wifi config menu item anywhere. Is there anything I am missing?
I have reverted back now to 15.7.0 as the camera image was also very dark and rotated without any reason

**Reply by SybexX** [reply]
@turbolooser sorry, there was another entry missing ^^
I updated the link above, this version should work.

**Comment by friedpa** [comment]
@SybexX Tested it and works fine. I even saw the AP Scan, which is missing in your pictures above :o)

One small thing: "Scan Wifi Ap" should be "Scan Wifi A**P**"

**Reply by SybexX** [reply]
@friedpa What I don't understand is why the percentage and the reception bar are displayed twice on the second network.
Do you happen to have a hidden network active?

**Comment by friedpa** [comment]
@SybexX Sorry,can´t tell you that. My Network consists of one FritzBox 6890 and a couple of Repeators (6 in total). Nothing is hidden.
But it seams that this issue comes up on different networks randomly:

![grafik](https://github.com/user-attachments/assets/584020ce-681c-4bb8-b8c2-ce538f0903cd)

**Reply by SybexX** [reply]
@friedpa Have you tried today's new version yet?
I have added some queries and hope to have solved this problem.

**Comment by friedpa** [comment]
@SybexX Schaut leider aus wie vorher....

![grafik](https://github.com/user-attachments/assets/e81be9dd-b853-40d9-a58c-e7d8770258a7)
![grafik](https://github.com/user-attachments/assets/16f63d00-6796-414b-b0ba-a0c186fba700)

**Reply by SybexX** [reply]
Kannst du bitte gucken, was die Konsole ausgibt?

![Unbenannt](https://github.com/user-attachments/assets/2851a752-6525-4ced-a548-c843ef7a6bc3)

**Comment by friedpa** [comment]
Ich habe keine Liste mit den SSIDs gefunden

![grafik](https://github.com/user-attachments/assets/4d944cbc-114c-4790-bd02-071438237d70)

**Comment by friedpa** [comment]
Jetzt mit dem Fehler...

![grafik](https://github.com/user-attachments/assets/2036870c-a13c-4248-9325-d5b918cb9ad3)

**Reply by SybexX** [reply]
Oder liegt es eventuell an deinem Browser, welchen benutzt du? 
Ich benutze Firefox und gelegentlich Microsoft Edge und bei beiden wird die JSON-Liste in der Konsole ausgegeben.
Du könntest auch "http://deine_IP/wifiscan" versuchen.

**Comment by friedpa** [comment]
Ich habe Firefox und Chrome probiert, beide gleich. Ich habe nun auch in dem anderen ESP32 die SW geladen.... da kommt folgendes:

![360080659-54835292-0772-490a-81c2-c34509507994](https://github.com/user-attachments/assets/ad147b11-9932-49d7-a0ba-7ee75c268a65)

Dieser ESP32 sieht nur zwei Netze, kann es daher sein, dass der andere ESP32, der jetzt mehr Netzte sieht, wie im Beispiel unten 9 Netze, ein Problem mit der Anzahl von Netzen hat?

![grafik](https://github.com/user-attachments/assets/695d14ab-4c22-44fd-abbf-33b7bf56e941)

EDIT: Was mir noch aufgefallen ist, bei dem ESP32 mit dem Fehler musste ich den Upload mindestens 5 Mal probieren, bei dem anderen ESP32 ohne Fehler ging der Upload sofort durch.....

**Reply by SybexX** [reply]
Die Anzahl ist in der Firmware auf 10 begrenzt, alle anderen Wifis werden eigentlich ignoriert, um keine Speicherprobleme zu bekommen. Hast du "http://deine_IP/wifiscan" ausprobiert?

**Comment by friedpa** [comment]
Der ESP32 mit dem Fehler ist jetzt tot,... Boot Loop....kann erst morgen wieder weitertesten...

**Comment by TheBestian7** [comment]
Hallo, danke für die Arbeit!! Hat mir viel Arbeit erspart :-)
Habe AI-on-the-edge-device__update__add_wifi_config_page_(a86e39b) installiert und ähnliches Bild.

![Screenshot_WiFi](https://github.com/user-attachments/assets/cbb36f30-de49-40ea-9fad-474ed6a62ae6)
![Screenshot_wifiscan](https://github.com/user-attachments/assets/3c0981cd-7852-4128-8bba-dd8c62befb18)

wifi1 auf dem Screenshot hat keine SSID, kann der "Fehler" daher kommen? 
LG
Sebastian

**Reply by TheBestian7** [reply]
Habe die WiFi Config nochmal aufgerufen und habe jetzt 2x den Eintrag der Signalstärke ohne Zuordnung und im wifiscan 2x eine leere SSID

**Reply by SybexX** [reply]
Das Wifi ohne Namen könnte ein verstecktes Netzwerk sein und verursacht den Fehler^^
Ich aktualisiere den Link oben immer so, dass er zur neuesten Version führt und dahinter steht das Datum und die Uhrzeit, zu der ich aktualisiert habe.

**Reply by TheBestian7** [reply]
Ja kann sein das die Nachbarn versteckte Netzwerke haben. Letztlich auch egal, wichtig ist das man sein WLAN ändern kann. Danke dafür :-)

**Reply by SybexX** [reply]
ich habe noch eine Änderung gemacht, jetzt sollte es ok sein. 

https://github.com/SybexX/AI-on-the-edge-device/actions/runs/10669878308

**Comment by friedpa** [comment]
@SybexX Wird dieses Feature in die neue Main Release aufgenommen?
Ich wäre dafür, auch wenn es keinen Password Schutz gibt.... Wer soll sich schon mit so einer Datei spielen.... Wenn der Nachbar den Verbrauch wissen will, braucht er nur den Zählerkasten öffnen....

**Reply by SybexX** [reply]
Soweit ich weiß, wollen die anderen Collaborators das nicht.
Dabei geht es nicht um den Zählerstand, sondern darum, dass man aus dem Web auf die wifi.ini zugreifen könnte bzw. auf die Wifi-Zagangsdaten.

**Comment by friedpa** [comment]
Was ist daran das Problem, wer soll so ein Gerät kappern? Und wenn er es tut, habe ich die Möglichkeit es abzustecken. Das Password eines WIFIs bekomme ich auch so heraus, da muss ich nicht erst in einer WLAN.ini nachschauen, abgesehen davon, dass das keiner weiß....
Die Akteure die jemanden infiltrieren wollen haben wirklich andere Möglichkeiten......

**Reply by SybexX** [reply]
hier kannst du sehen was noch ins nächste Release rein soll https://github.com/jomjol/AI-on-the-edge-device/labels/Release16.0.0

**Comment by friedpa** [comment]
Vielleicht nochmals an die "anderen" Developers :o) Es ist ein sehr praktisches Feature, man muss nicht an oft unmöglichen Orten das Gerät wieder ausbauen, wenn man sein Netzwerk umkonfiguriert, oder das Passsword tauschen muss...
Wenn man bereits jemanden in seinem Netzwerk hat, ist das dann wirklich eh schon egal, ob er auch sehen kann welchen Wassser-, Gas-, ... was auch immer Verbrauch einer hat... dann hat man wirklich eher andere Probleme....

**Reply by SybexX** [reply]
@friedpa Hast du schon überprüft, ob das Problem mit der neuesten Version behoben wurde?
![359975546-2036870c-a13c-4248-9325-d5b918cb9ad3](https://github.com/user-attachments/assets/4167bcc0-c981-4f21-bc81-aa56a552b3aa)

**Reply by TheBestian7** [reply]
Bei mir ist es behoben.

**Comment by friedpa** [comment]
Sorry, habe nicht reported.... 

![grafik](https://github.com/user-attachments/assets/85aa1dbd-bc5b-4c80-a54c-2d444b31e846)

**Comment by TheBestian7** [comment]
@SybexX Nachdem ich meine WLAN Config geändert habe bin ich wieder zurück auf die "normale" v.15.7.0 gegangen weil das Bild so dunkel war. Als ich in der Config war und was geändert habe wurde ich darauf hingewiesen das die Image Size nicht angegeben ist und ich das doch bitte anpassen soll bevor ich speichere. Das Feld ist ausgegraut, man muss die Expert Parameters aktivieren und dann in der Config.ini Editor bei ImageSize und ImageQuality das Semikolon aus der Zeile entfernen und Werte eintragen.

Weiter oben habe ich schonmal gelesen das jemand sagte bei ihm war das Bild so dunkel mit deiner Firmware. 
Solltest du die Version noch weiter entwickeln wollen wäre das imho ein Punkt den du dir anschauen könntest. Danke erneut für deine Arbeit, LG Sebastian

![image](https://github.com/user-attachments/assets/540ca89a-8d6d-48b5-83fd-54004cb60cd5)

![image](https://github.com/user-attachments/assets/f0f82730-905d-4304-84ff-2716c98e9ee6)

**Reply by SybexX** [reply]
Du mußt mindestens ein mal speichern und dann werden die fehlenden Einträge in die config.ini geschrieben.
Um das Bild heller zu kriegen, mußt du mit den Kamera-Parametern rumspielen oder meine nehmen: https://github.com/SybexX/AI-on-the-edge-device/blob/test/sd-card/config/config.ini

![Unbenannt](https://github.com/user-attachments/assets/75f713f3-cc13-45a4-95ac-c47fa71d948d)

Mit erhöhen von CamGainceiling kannst du das Bild heller machen und dadurch die LEDIntensity reduzieren.

Das Problem mit den ausgegrauten Einträgen ist im aktuellen rolling, in der Version 15.7.0 und Anderen Versionen vorhanden und muß noch gefixt werden^^

Alle Kameraparameter wurden umbenannt und ImageSize/ImageQuality gehört zur Version 15.7.0.
Wenn die Parameter nicht in der config.ini vorhanden sind, werden die Standardparameter von der Kamera verwendet.

**Comment by SybexX** [comment]
aktuelles rolling + wifi config + basic_auth + IgnoreAllNaNs + Freenove_WROVER support + Übersrbeitung der Recognition Seite:
https://github.com/SybexX/AI-on-the-edge-device/actions/runs/10669878308

![Unbenannt](https://github.com/user-attachments/assets/50cbb1cf-e249-4ea2-942a-c9c978fbff09)
![Unbenannt1](https://github.com/user-attachments/assets/501d05a0-fc5b-45c8-bdab-3acd80878771)
![363519702-414c7cf0-db4a-4ba4-a647-b14b512ac0e2](https://github.com/user-attachments/assets/cce38fba-aa3a-4d89-b802-7061e2d6bf43)

**Reply by SybexX** [reply]
Neue Version zum testen^^
In der ersten Version konnte man aus dem Web aus auf die wlan.ini zugreifen, in dieser Version ist dies nicht mehr möglich. 
Das aktuelle WLAN-Passwort wird jetzt weder an die Web-Benutzeroberfläche gesendet noch dort angezeigt.
Wenn der ESP32 neue Daten erhält(ssid, pw, ip.....), speichert er diese eigenständig in die wlan.ini.

https://github.com/SybexX/AI-on-the-edge-device/actions/runs/10692355159

**Comment by friedpa** [comment]
Der Link funktioniert leider nicht....

**Reply by SybexX** [reply]
welchen Link meinst du? bei mir gehen noch alle

**Comment by friedpa** [comment]
https://github.com/SybexX/AI-on-the-edge-device/actions/runs/10689763493

geht bei mir nicht (404 Error).....

Edit: Meine Schuld,.... war nicht eingeloggt....

**Comment by friedpa** [comment]
Wenn das so gewollte ist, dann ok...

![grafik](https://github.com/user-attachments/assets/6cc6d3f6-2118-4357-b5a4-523776c5c31d)

**Reply by SybexX** [reply]
ja, das ist gewollt. Denn wenn man das PW anzeigen möchte, muss man es ja an die Website senden und das ist unsicher^^

**Comment by friedpa** [comment]
Na dann, alles OK. SW läuft bei mir.....

**Reply by SybexX** [reply]
Dies soll ein Versuch sein, die Andere dazu zu bewegen, dies doch noch in das rolling aufzunehmen ^^

**Comment by friedpa** [comment]
Meine Unterstützung hast Du! 
Begründung: Ich wechsle regelmäßig mein Password im Home-WLAN. Jetzt könnte ich, ohne in den Wasserschacht zu klettern, des ESP auf das Gast-WLAN umhängen, das Password im Home-WLAN änderm und dann den ESP in das Home-WLAN switchen....
Wäre für mich, und vielleicht auch für den einen, oder andern, eine Erleichterung.... 

You have my support! 
Reason: I regularly change my password in the Home-WLAN, Now I could switch the ESP to the Guest-WLAN without climbing into the water shaft, change the password in the Home-WLAN and then switch the ESP to the Home-WLAN.... Would be a relief for me, and maybe also for one or the other....

**Comment by SybexX** [comment]
@friedpa neue Version zum testen^^
Das Pw ist auf der Seite wieder sichtbar, wird jedoch mit einer leichten Verschlüsselung vom/zum ESP32 übertragen.
[AI-on-the-edge-device__update__test1_(bySybexX).zip](https://github.com/user-attachments/files/17119434/AI-on-the-edge-device__update__test1_.bySybexX.zip)

**Comment by friedpa** [comment]
Funktioiert. Er zeigt auch das richtige Password an. Ob die Verschlüsselung funktioniert kann ich aber nicht feststellen...

![grafik](https://github.com/user-attachments/assets/b8affac8-514e-4867-82e4-1191550f2007)

**Reply by SybexX** [reply]
Wenn das Password richtig angezeigt wird und ein neues Password auf der SD richtig gespeichert wird, dann funktioniert die Verschlüsselung richtig.

**Reply by SybexX** [reply]
@friedpa neue Version zum testen: 
[AI-on-the-edge-device__update__test1_(bySybexX).zip](https://github.com/user-attachments/files/17119429/AI-on-the-edge-device__update__test1_.bySybexX.zip)

Mit dieser Version wird jetzt auch das Passwort auf der SD verschlüsselt.
Der Esp liest das Pw von der SD und wenn vor dem Pw ``**##**`` nicht vorhanden ist, verschlüsselt er das Pw und speichert es auf der SD ab.
zB.: aus    ``password = "test Pw" ``   wird dann    ``password = "**##**test Pw verschlüsselt"``

**Comment by friedpa** [comment]
Geht bei mir erst nächsten Montag, bin auf Kurzurlaub :o)

**Comment by friedpa** [comment]
Ich habe mich remote auf mein Testsystem verbunden und den Update doch gemacht.... :o)
Upgrade funktionierte soweit. Meine Frau hat dann die SD Karte rausgenommen und in meinen SD Card Reader gegeben.... Das Password ist verschlüsselt in der WLAN.ini..... SD Card zurück im Test System hat er gebooted... hängt aber in der Initiate Phase... das ist aber schon öfters im Testsystem passiert. Kann aber jetzt erst am Montag weiter machen....

**Reply by SybexX** [reply]
Es kann an der SD liegen, ich hatte dieses Problem auch mal, als ich 8gb Sandisk benutzte.
Jetzt benutze ich immer billige chinesische Karten und habe seitdem keine Probleme mehr^^
Ob es an der Firmware oder an der SD liegt, lässt sich am einfachsten testen, indem man das unverschlüsselte Passwort in die wlan.ini auf der SD einträgt und ausprobiert, ob der ESP wieder normal bootet.

**Reply by SybexX** [reply]
neue Version zum testen: [AI-on-the-edge-device__update__test1_(bySybexX).zip](https://github.com/user-attachments/files/17119424/AI-on-the-edge-device__update__test1_.bySybexX.zip)
In dieser Version werden alle Passwörter(von Wifi+mqtt+influxdb....) auf der SD verschlüsselt.

eine weitere Version zum testen: 
[AI-on-the-edge-device__update__test1_(bySybexX8).zip](https://github.com/user-attachments/files/17271710/AI-on-the-edge-device__update__test1_.bySybexX8.zip)
[AI-on-the-edge-device__update__test1_(bySybexX9).zip](https://github.com/user-attachments/files/17336736/AI-on-the-edge-device__update__test1_.bySybexX9.zip)
damit sollte der Seitenaufbau schneller sein, insbesondere die Data Graph Seite.

**Reply by mmmgggttt** [reply]
Hello @SybexX  many thanks for your contribution. I checked your repo, cannot find the latest update.zip for OTA aligned with the latest release of the main branch. Can you please tell me where i can find it ready to OTA to my device ?thank you Marco
